docs/quickstart/run_example_quickstart.md

```diff
@@ -27,116 +27,360 @@ under the License.
+        <artifactId>flink-connector-wikiedits_2.10</artifactId>
+        <version>${flink.version}</version>
+    </dependency>
+</dependencies>
+{% endhighlight %}
+
+Notice the `flink-connector-wikiedits_2.10` dependency that was added.
+
+## Writing a Flink Program
+
+It's coding time. Fire up your favorite IDE and import the Maven project or open a text editor and
+create the file `src/main/wikiedits/WikipediaAnalysis.java`:
+
+{% highlight java %}
+package wikiedits;
+
+public class WikipediaAnalysis {
+
+    public static void main(String[] args) throws Exception {
+
+    }
+}
+{% endhighlight %}
+
+I admit it's very bare bones now but we will fill it as we go. Note, that I'll not give
+import statements here since IDEs can add them automatically. At the end of this section I'll show
+the complete code with import statements if you simply want to skip ahead and enter that in your
+editor.
+
+The first step in a Flink program is to create a `StreamExecutionEnvironment`
+(or `ExecutionEnvironment` if you are writing a batch job). This can be used to set execution
+parameters and create sources for reading from external systems. So let's go ahead, add
+this to the main method:
+
+{% highlight java %}
+StreamExecutionEnvironment see = StreamExecutionEnvironment.getExecutionEnvironment();
+{% endhighlight %}
+
+Next, we will create a source that reads from the Wikipedia IRC log:
+
+{% highlight java %}
+DataStream<WikipediaEditEvent> edits = see.addSource(new WikipediaEditsSource());
+{% endhighlight %}
+
+This creates a `DataStream` of `WikipediaEditEvent` elements that we can further process. For
+the purposes of this example we are interrested in determining the number of added or removed
+bytes that each user causes in a certain time window, let's say five seconds. For this we first
+have to specify that we want to key the stream on the user name, that is to say that operations
+on this should take the key into account. In our case the summation of edited bytes in the windows
+should be per unique user. For keying a Stream we have to provide a `KeySelector`, like this:
+
+{% highlight java %}
+KeyedStream<WikipediaEditEvent, String> keyedEdits = edits
+    .keyBy(new KeySelector<WikipediaEditEvent, String>() {
+        @Override
+        public String getKey(WikipediaEditEvent event) {
+            return event.getUser();
+        }
+    });
+{% endhighlight %}
+
+This gives us the same Stream of `WikipediaEditEvent` that has a `String` key that is the user.
+We can now specify that we want to have windows imposed on this stream and compute some
+result based on elements in these windows. We need to specify a window here because we are
+dealing with an infinite stream of events. If you want to compute an aggregation on such an
+infinite stream you never know when you are finished. That's where windows come into play,
+they specify a time slice in which we should perform our computation. In our example we will say
```

Comment:
Yes, that's a tricky one. I don't think we have an explanation of why windows are required anywhere in the doc. I also don't think however, that this guide is the right place, since I want this to be very concise.